sdk/search/Azure.Search.Documents/src/Models/SearchQueryType.Serialization.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System;
+
+namespace Azure.Search.Documents.Models
+{
+    /// <summary>
+    /// TODO: Remove this custom serializer when enabling Semantic search
+    /// </summary>
+    internal static partial class SearchQueryTypeExtensions
+    {
+        public static string ToSerialString(this SearchQueryType value) => value switch
+        {
+            SearchQueryType.Simple => "simple",
+            SearchQueryType.Full => "full",
+            // SearchQueryType.Semantic => "semantic",
+            _ => throw new ArgumentOutOfRangeException(nameof(value), value, "Unknown SearchQueryType value.")
```

Comment:
Nit: I'd use `nameof(SearchQueryType)` in an interpolated string (the compiler will optimize to a const string, since `nameof(SearchQueryType)` is a const) just in case this is refactored before GA. Nice and clean that way.

Reply:
Heh, I had it that way in an older version. I'll update. Thanks for the comment.